src/crypting/Main.java

```diff
@@ -0,0 +1,36 @@
+package crypting;
+
+import crypting.strategy.ShiftCipher;
+import crypting.strategy.UnicodeCipher;
+
+public class Main {
+
+    public static void main(String[] args) {
+        CliParser parser = new CliParser(args);
+        Cryptor cryptor;
+
+        String outFilePath = parser.optionOf("-out");
+        String inFilePath = parser.optionOf("-in");
+        String mode = parser.optionOf("-mode");
+        String algorithm = parser.optionOf("-alg");
+        String data = parser.optionOrDefault("-data", "");
+        int key = Integer.parseInt(parser.optionOrDefault("-key", "0"));
+
+        cryptor = new Cryptor("unicode".equals(algorithm) ?
+                new UnicodeCipher() :
+                new ShiftCipher()
+        );
+
+        if (data.isEmpty() && inFilePath != null)
```

Comment:
Complex logic, let's simplify it. 
Correct me if I am wrong, we have both options how to pass data to your service

- Pass by `-data` argument 
- Pass by `-in` argument (data will be inside the file)

So if data is not empty, then we don't care about `-in` argument, right?

Also, data and inFilePath variables are String, but one could be empty and other could be null. Think about it 👌

Reply:
But how to act when there will be `-data "" -in "input.txt"?`

Reply:
Is it correct way to pass such data to the service? I think then we need to have priority between these arguments or throw exception in that case. What do you think?

Reply:
Updated. I have added data validation method and simplified it as we discussed